src/test/java/org/assertj/core/internal/throwables/Throwables_assertHasMessageContaining_Test.java

```diff
@@ -46,11 +53,24 @@ public void should_fail_if_actual_is_null() {
   @Test
   public void should_fail_if_actual_has_message_not_containing_with_expected_description() {
     AssertionInfo info = someInfo();
+    String content = "expected description part";
+    try {
+      throwables.assertHasMessageContaining(info, actual, content);
+      fail("AssertionError expected");
+    } catch (AssertionError err) {
+      verify(failures).failure(info, shouldContain(actual.getMessage(), content), actual.getMessage(), content);
+    }
+  }
```

Comment:
let's rewrite these tests with java 8 love like this:
```java
import static org.assertj.core.util.AssertionsUtil.expectAssertionError;

    // GIVEN
    AssertionInfo info = someInfo();
    String content = "expected description part";
    // WHEN
    expectAssertionError(() -> throwables.assertHasMessageContaining(info, actual, content));
    // THEN
    verify(failures).failure(info, shouldContain(actual.getMessage(), content"));
```
I think this is much more readable and elegant than `try catch` (due to the tests being written for Java 7 at the time).